pinecone/client.go

```diff
@@ -59,6 +143,49 @@ func NewClient(in NewClientParams) (*Client, error) {
 	return NewClientBase(NewClientBaseParams{Headers: clientHeaders, Host: in.Host, RestClient: in.RestClient, SourceTag: in.SourceTag})
 }
 
+// NewClientBase creates and initializes a new instance of Client with custom authentication headers,
+// allowing users to authenticate in ways other than passing an API key.
+//
+// This function requires an input parameter of type NewClientBaseParams, which includes:
+//   - Headers: A map of additional HTTP headers to include in the API request.
+//     "Authorization" and "X-Project-Id" headers are required.
```

Comment:
Is `"X-Project-Id"` required? I'm not sure if it actually is. We may want to be more general rather than specifying how they could provide authorization. If they wanted to pass the API key themselves as a header they'd need to use `Api-Key` as the key anyways.

I'd maybe just say "If using NewClientBase you must provide authorization through Headers or the RestClient." Would that work?